src/golang/lib/engine/aq_engine.go

```diff
@@ -797,12 +821,25 @@ func (eng *aqEngine) execute(
 					if opExecMode == operator.Publish {
 						err = dagOp.PersistResult(ctx)
 						if err != nil {
-							return errors.Wrapf(err, "Error when finishing execution of operator %s", op.Name())
+							err = errors.Wrapf(err, "Error when finishing execution of operator %s", op.Name())
+							return err
 						}
 					}
 				}
 
+				hasNotification = true
+				notificationLevel = mdl_shared.ErrorNotificationLevel
+				if execState.Error != nil {
```

Comment:
In what cases do we not write to execState.Error? I thought it was expected that this was always written to on error.

Reply:
This is just a defensive check as this field is a pointer and we are accessing subfields in the block. This way if anything is wrong, the nil pointer access won't kill the whole process